File: pgm/ABC_Visual_Basic/ABC_131_B.vb
' (N * (L + (L + N - 1))) / 2
' (N * (2 * L + N - 1)) / 2
' Imports System.Collections.Generic
' Imports System.Globalization
' Imports System.IO
' Imports System.Linq
' Imports System.Text
' Imports System.Text.RegularExpressions

Class Main
    Public Shared Sub Main()
        Dim N, L As Integer
        Dim arr() As String
        arr = Split(Console.ReadLine())
        N = arr(0) : L = arr(1)
        Dim ans As Integer = (N * (2 * L + N - 1)) / 2
        If 0 < L
            ans -= L
        ElseIf L + N - 1 < 0
            ans -= L + N - 1
        End If
        Console.WriteLine(ans)
        Return
    End Sub
End Class

File: pgm/ABC_C#/ABC_131_B.cs
// (N * (L + (L + N - 1))) / 2
// (N * (2 * L + N - 1)) / 2
using System;
// using System.Collections.Generic;
// using System.Globalization;
// using System.IO;
// using System.Linq;
// using System.Text;
// using System.Text.RegularExpressions

class Program
{
    static void Main()
    {
        string[] arr = Console.ReadLine().Split();
        int N, L;
        N = int.Parse(arr[0]);
        L = int.Parse(arr[1]);
        int ans = (N * (2 * L + N - 1)) / 2;
        if (0 < L) ans -= L;
        else if (L + N - 1 < 0) ans -= L + N - 1;
        Console.WriteLine(ans);
        return;
    }
}